foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/asm/ASMFactory.java

```diff
@@ -26,25 +27,45 @@
 import java.security.AccessController;
 import java.security.PrivilegedAction;
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.regex.Pattern;
 
 public class ASMFactory {
 
-    private final static SessionLog LOG = AbstractSessionLog.getLog();
-
+    // This block must be first - begin
     public final static String ASM_SERVICE_ECLIPSELINK = "eclipselink";
     public final static String ASM_SERVICE_OW2 = "ow2";
+
     private final static String ASM_OW2_CLASS_VISITOR = "org.objectweb.asm.ClassVisitor";
     private final static String ASM_ECLIPSELINK_CLASS_VISITOR = "org.eclipse.persistence.internal.libraries.asm.ClassVisitor";
 
-    //Should be changed in case of ASM upgrade
-    public final static int ASM_API_SELECTED = Opcodes.valueInt("ASM9");
-    public final static int JAVA_CLASS_VERSION = Opcodes.valueInt("V1_8");
+    private final static String ASM_OPCCODES_OW2 = "org.objectweb.asm.Opcodes";
+    private final static String ASM_OPCCODES_ECLIPSELINK = "org.eclipse.persistence.internal.libraries.asm.Opcodes";
+
+    private final static Map<String, String> ASM_OPCCODES_MAP = new HashMap<>();
```

Comment:
Duplicit with `org.eclipse.persistence.asm.Opcodes#ASM_OPCCODES_MAP`. Maybe change scope and use this one only.